### PR TITLE
fix: reject undersized domain ghost widths

### PR DESCRIPTION
**Pull request by atulcthakur**
# ALCHEMI Toolkit Pull Request

## Description

An explicit `ghost_width` could be smaller than `cutoff + skin`. This can exclude atoms still required by the neighbor list and produce incorrect energies or forces.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Changes Made

- Reject explicit `ghost_width < cutoff + skin`.
- Keep `ghost_width=None` defaulting to `cutoff + skin`.
- Allow exact equality.
- Add tests for rejected and valid boundary values.

- `nvalchemi/distributed/config.py`
- `test/distributed/test_config.py`

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [ ] I have performed a self-review of my code
- [ ] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

**Comment by atulcthakur** [comment]
/ok to test https://github.com/NVIDIA/nvalchemi-toolkit/pull/152/commits/c5abe6021be4f7c0c3b271fc5ee22254e338ac63

File: nvalchemi/distributed/config.py [hunk 1]
```diff
@@ -136,6 +137,16 @@ def _grid_dims_positive(
             raise ValueError(f"grid_dims entries must be >= 1, got {v}")
         return v
 
+    @model_validator(mode="after")
+    def _ghost_width_covers_cutoff_and_skin(self) -> DomainConfig:
+        required_width = self.cutoff + self.skin
+        if self.ghost_width is not None and self.ghost_width < required_width:
```
[the remark this answers is not in the document]
Comment (laserkelvin) [reply nvalchemi/distributed/config.py:143]: Confirmed this locally

File: nvalchemi/distributed/config.py
Comment (laserkelvin) [inline nvalchemi/distributed/config.py:129]: I think we should also validate against this as well:

```python
hysteresis <= skin / 2
```

**Review by laserkelvin — APPROVED** [review]
I've left a comment, but the change itself is good with me

File: nvalchemi/distributed/config.py
Comment (atulcthakur) [reply nvalchemi/distributed/config.py:129]: I was about to address this and the P1 from greptile.
